src/basic/util.c

```diff
@@ -832,6 +832,62 @@ uint64_t physical_memory_scale(uint64_t v, uint64_t max) {
         return r;
 }
 
+uint64_t system_tasks_max(void) {
+
+#if SIZEOF_PID_T == 4
+#define TASKS_MAX ((uint64_t) (INT32_MAX-1))
+#elif SIZEOF_PID_T == 2
+#define TASKS_MAX ((uint64_t) (INT16_MAX-1))
+#else
+#error "Unknown pid_t size"
+#endif
```

Comment:
Just a random thought when reading this, ignore if it's bogus:

Not sure if cpp is clever enough to evaluate this statically, but would something like this work?

    #define TASKS_MAX ((uint64_t) (2 << (SIZEOF_PID_T * 8) - 1)

(could be that this causes  an overflow as 2 << 64 == 0,  though).